CRM/Civioffice/Form/DocumentRenderer/Settings.php

```diff
@@ -61,7 +61,7 @@ public function preProcess()
         // Require type for adding.
         if ($this->_action & (CRM_Core_Action::ADD)) {
             if (!$type = CRM_Utils_Request::retrieve('type', 'Alphanumeric', $this)) {
-                throw new Exception(E::ts('Missing Document Renderer type.'));
+                throw new Exception(E::ts('Missing Document Renderer.'));
```

Comment:
This was actually correct, it's about the renderer type.

Reply:
If this is not a string visible for users, I'm wondering if a translation is really necessary. Otherwise I think it should be without `type` (as it already was in the German translation), even though it's technically not precise...

Reply:
Sure, whether to translate exception messages in the first place is debatable; I tend to have them translated in CiviCRM extensions, as they will show up in user-facing logs in the UI.

But would you agree that exception messages should be as technically precise as possible? This form takes care of adding/editing renderers itself and needs a renderer *type* name for adding one, which when that is missing is what the exception message complains about, isn't it?

Btw, this changed recently, having document renderers and document renderer types be separate things now. Maybe this was confusing you?

Reply:
We agree that exception messages should be precise in general, so I'm okay with adding *type* again, even though users wouldn't know what it means.

The confusion potential of *renderer* and *renderer type* is a topic for itself ;-)